File: 263-ugly-number/263-ugly-number.java
class Solution {
    public boolean isUgly(int n) {
        if(n <= 0){
            return false;
        }
        
        if(n == 1){
            return true;
        }
        
        for(int f : new int[]{2, 3, 5}){
            if((n % f) == 0 && this.isUgly(n / f)){
                return true;
            }
        }
        
        return false;
    }
}

File: 263-ugly-number/263-ugly-number.py
class Solution:
    def isUgly(self, n: int) -> bool:
        if n <= 0:
            return False
        
        if n == 1:
            return True
        
        for f in [2, 3, 5]:
            nn, r = divmod(n, f)
            if r == 0 and self.isUgly(nn):
                return True
        return False